### PR TITLE
7.0.2-devel

### PR DESCRIPTION
**Pull request by maciekpac**
This change is [Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)

### CODE REVIEW

**Review by bratpiorka — CHANGES_REQUESTED** [review]
> Reviewed 24 of 24 files at r1, 8 of 8 files at r2, 11 of 11 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: all files reviewed, 1 unresolved discussion (waiting on @maciekpac and @PatKamin)

___
*[`src/adlist.h` line 81 at r1](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-N6O2xUT9pZlBN5MFt68:-N6O2xUT9pZlBN5MFt69:b-kauf1d) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/1100e49c45b1faf45ac22313e29c96a2c8cf50b2/src/adlist.h#L81)):*
> ```objc
> list *listAddNodeHead(list *list, void *value);
> list *listAddNodeHeadDRAM(list *list, void *value);
> list *listAddNodeTail(list *list, void *value);
> ```

why there is no tail DRAM version?

**Review by bratpiorka — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: all files reviewed, 1 unresolved discussion (waiting on @maciekpac and @PatKamin)

**Review by maciekpac — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: all files reviewed, 1 unresolved discussion (waiting on @bratpiorka and @PatKamin)

___
*[`src/adlist.h` line 81 at r1](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-N6O2xUT9pZlBN5MFt68:-N6OaAR12Sdv0VsTEuCL:b-ahw0fb) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/1100e49c45b1faf45ac22313e29c96a2c8cf50b2/src/adlist.h#L81)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

why there is no tail DRAM version?
</blockquote></details>

I suppose this was not needed, maybe we only need to add DRAM nodes at head ?

6.2.5-devel does not have this function either;

If we want to investigate it, I think we should set up a separate task/do it later - I'm only porting code from old branches here

**Review by bratpiorka — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PatKamin)

___
*[`src/adlist.h` line 81 at r1](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-N6O2xUT9pZlBN5MFt68:-N6RccwdA3JeJWCWE1z1:b-4obeei) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/1100e49c45b1faf45ac22313e29c96a2c8cf50b2/src/adlist.h#L81)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

I suppose this was not needed, maybe we only need to add DRAM nodes at head ?

6.2.5-devel does not have this function either;

If we want to investigate it, I think we should set up a separate task/do it later - I'm only porting code from old branches here
</blockquote></details>

ok, agree - I will add separate task for that

**Review by bratpiorka — APPROVED** [review]
> Reviewed 2 of 15 files at r4, 3 of 8 files at r5, 11 of 11 files at r6, 1 of 1 files at r9, 1 of 1 files at r10, 2 of 2 files at r12, 1 of 1 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PatKamin)

**Review by PatKamin — CHANGES_REQUESTED** [review]
> Reviewed 1 of 24 files at r1, 1 of 11 files at r6, 1 of 1 files at r13.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: all files reviewed, 1 unresolved discussion (waiting on @maciekpac)

___
*[`-- commits` line 4 at r13](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-N6gR94D4icfWZ3oAbgE):*
v1.14.0

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     - name: Install test dependencies
       run: sudo apt install  tcl8.6 tclx
 
-    - name: Build memKeydB
+    - name: Build TieredMemDB
```
Comment (PatKamin) [inline .github/workflows/ci.yml:32]: Please, merge this commit with the one that introduces memkind in github actions. No reason for keeping a fix commit with the commit introducing changes in the very same PR.

**Review by maciekpac — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: all files reviewed, 1 unresolved discussion (waiting on @PatKamin)

___
*[`-- commits` line 4 at r13](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-N6gR94D4icfWZ3oAbgE):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

v1.14.0
</blockquote></details>

Done.

**Review by PatKamin — COMMENTED** [review]
> Reviewed 5 of 35 files at r14.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: 0 of 43 files reviewed, all discussions resolved (waiting on @bratpiorka and @PatKamin)

**Review by maciekpac — COMMENTED** [review]
Done

> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-:-N6hQgur967QDk79K7YR:bb6un1j)* status: 0 of 43 files reviewed, all discussions resolved (waiting on @bratpiorka and @PatKamin)

**Review by bratpiorka — APPROVED** [review]
> Reviewed 15 of 24 files at r21, 28 of 28 files at r33, 11 of 11 files at r34, 12 of 12 files at r35, 10 of 10 files at r36, 11 of 11 files at r37, 18 of 18 files at r38, 10 of 10 files at r39, 11 of 11 files at r40, 3 of 3 files at r41, 1 of 1 files at r42, 10 of 10 files at r43, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @maciekpac)

**Review by PatKamin — APPROVED** [review]
:lgtm:

> Reviewed 15 of 24 files at r21, 28 of 28 files at r33.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/29#-:-N6r2SXM2GJcczNCTsBg:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @maciekpac)

**Pull request merged**